### PR TITLE
Fixes #20855 - View SP content with view_capsule_content permission

### PR DESCRIPTION
**Pull request by sean797**

### CODE REVIEW

**Comment by jlsherrill** [comment]
@sean797 mind fixing the commit message? (missing a dash)

**Comment by sean797** [comment]
Ah @jlsherrill you mean the titile :) updated!

File: app/controllers/katello/api/v2/capsule_content_controller.rb [hunk 1]
```diff
@@ -91,7 +91,7 @@ def respond_for_lifecycle_environments_index(environments)
     end
 
     def find_capsule
-      @capsule = SmartProxy.unscoped.authorized(:manage_capsule_content).find(params[:id])
+      @capsule = SmartProxy.unscoped.authorized(:view_capsule_content).find(params[:id])
```
Comment (jlsherrill) [inline app/controllers/katello/api/v2/capsule_content_controller.rb]: So i think for certain actions we still need the 'manage_capsule_content' option. For cancel_sync and sync, add/remove_lifecycle_environments, we'd want manage_capsule_content permissions I would assume, but for everything else, we'd want view_capsule_content.  You'll probably need a 'find_capsule' method and a 'find_editable_capsule' method here, and then modify the before_action to call a different one for the different actions.

Comment (jlsherrill) [reply app/controllers/katello/api/v2/capsule_content_controller.rb]: And this ignores teh fact that add/remove_lifecycle_environemnts likely use different permissions between the UI and api.  We really need to just collapse all these permissions into the smart_proxy permissions, but that is another bug for another day :)   I'd just try to perserve the manage_capsule_content permission on those 4 actions (sync, cancel_sync, add/remove_lifecycle_envs) for now

**Comment by sean797** [comment]
Thanks for reviewing @jlsherrill :-) Just updated..

**Comment by sean797** [comment]
ping @jlsherrill can we get this merged for 3.4.6 ?

**Pull request merged**